aten/src/ATen/native/quantized/cpu/ACLUtils.cpp

```diff
@@ -0,0 +1,310 @@
+          weight_dim_1,
+          weight_scale,
+          weight_offset,
+          weight_ptr,
+          bias_ptr,
+          cache_key) {
+  int64_t m = std::get<static_cast<int>(QuantMatmulCacheKeyIndex::M)>(key);
+
+  auto src_q_tensor_info = arm_compute::TensorInfo(
+      arm_compute::TensorShape(weight_dim_0, m),
+      1,
+      // ACL dyanamically quantized matmuls only support (signed) int8_t
+      arm_compute::DataType::QASYMM8_SIGNED,
+      // TODO: setting the initial offset value to int8_t max instead of zero,
+      // because ACL currently skips MatrixBReduction calculation if the
+      // source offset at configuration time is zero. This is fixed by this
+      // PR: https://review.mlplatform.org/c/ml/ComputeLibrary/+/12820/8 This
+      // will be set to the actual src offset value at runtime.
+      arm_compute::QuantizationInfo(
+          /*scale=*/1.0,
+          /*offset=*/std::numeric_limits<int8_t>::max(),
+          /*is_dynamic=*/true));
+  src_q_tensor_info.set_are_values_constant(false);
+
+  auto src_tensor_info = arm_compute::TensorInfo(
+      arm_compute::TensorShape(weight_dim_0, m), arm_compute::Format::F32);
+  src_tensor_info.set_are_values_constant(false);
+
+  auto dst_tensor_info = arm_compute::TensorInfo(
+      arm_compute::TensorShape(weight_dim_1, m), arm_compute::Format::F32);
+
+  src_q_tensor.allocator()->init(src_q_tensor_info);
+  src_tensor.allocator()->init(src_tensor_info);
+  dst_tensor.allocator()->init(dst_tensor_info);
+
+  src_q_tensor_orig_ =
+      at::empty({m, weight_dim_0}, at::device(c10::kCPU).dtype(c10::kQInt8));
+  // allocate/import memory
+  src_q_tensor.allocator()->import_memory(src_q_tensor_orig_.data_ptr());
+
+  if (relu_info_.has_value()) {
+    relu = arm_compute::NEActivationLayer();
+  }
+}
+
+DynamicQuantMatmul::~DynamicQuantMatmul() {
+  // this will not free memory, it will just tell ACL that we're no longer
+  // using the pointer
+  src_q_tensor.allocator()->free();
+}
+
+arm_compute::Status DynamicQuantMatmul::validate() {
+  if (relu_info_.has_value()) {
+    auto relu_status = arm_compute::NEActivationLayer::validate(
+        dst_tensor.info(), dst_tensor.info(), relu_info_.value());
+    if (relu_status.error_code() != arm_compute::ErrorCode::OK) {
+      return relu_status;
+    }
+  }
+  auto quant_status = arm_compute::NEQuantizationLayer::validate(
+      src_tensor.info(), src_q_tensor.info());
+  if (quant_status.error_code() != arm_compute::ErrorCode::OK) {
+    return quant_status;
+  }
+  return arm_compute::NEGEMMLowpMatrixMultiplyCore::validate(
+      src_q_tensor.info(),
+      wei_q_tensor_.info(),
+      bia_tensor_.has_value() ? bia_tensor_.value().info() : nullptr,
+      dst_tensor.info(),
+      gemm_info_);
+}
+
+void DynamicQuantMatmul::configure() {
+  quant.configure(&src_tensor, &src_q_tensor);
+  gemm.configure(
+      &src_q_tensor,
+      &wei_q_tensor_,
+      bia_tensor_.has_value() ? &bia_tensor_.value() : nullptr,
```

Comment:
Nit
```suggestion
      bia_tensor_.value_or(nullptr),
```

Reply:
Sorry, I've made a mistake there, you'll need to keep the original logic, as you are passing pointer to tensor or null